src/Components/AppTile/AppTile.tsx

```diff
@@ -21,14 +23,35 @@ const AppTile = ({
     (state) => state.pathFinding.selectedTileType
   );
   const graph = useAppSelector((state) => state.pathFinding.graph);
+  const path = useAppSelector((state) => state.pathFinding.path);
 
   const handleClick = () => {
     if (!disabled && tileType && position) {
       dispatch(setGraphTile({ i: position.i, j: position.j, tileType }));
     }
   };
-  const bgcolor =
-    graph && position ? COLORS[graph[position.i][position.j]] : color;
+
+  const getBGColor = () => {
+    // Check if position exists
+    if (position) {
+      // Check if position is part of the path
+      const isPositionInPath = path?.some((element) => {
+        if (element.i == position.i && element.j == position.j) {
+          return true;
+        }
+
+        return false;
+      });
+      // Return the path color
+      if (isPositionInPath) {
+        return COLORS.PATH_TILE;
+      }
+    }
+    // Check if graph and position exists, if so return the position tile color, otherwise return empty tile color
```

Comment:
This is simply wrong.

Reply:
what is wrong?

Reply:
The comment you wrote incorrectly sums up the logic of that line, specifically the "otherwise" part.